url/src/lib.rs

```diff
@@ -2313,6 +2313,12 @@ impl Url {
     /// ```
     #[allow(clippy::result_unit_err, clippy::suspicious_operation_groupings)]
     pub fn set_scheme(&mut self, scheme: &str) -> Result<(), ()> {
+        // If the given scheme contains leading or trailing C0 controls,
+        // we'll ignore the set_scheme operation.
+        if scheme.trim_matches(|ch| ch <= ' ').len() != scheme.len() {
```

Comment:
There are two problems here:
1. I think we should early exit if the scheme `contains` a Non-tab/newline C0 controls - not just leading or trailing
2. This would still match tabs and newlines.

Please add a test for one of these chars in the middle of the scheme.
Also a test that setting it to `https\n` still works.